connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraCoGroupedRDD.scala

```diff
@@ -239,9 +239,8 @@ class CassandraCoGroupedRDD[T](
     scanRDDs.maxBy(_.partitions.length).partitions
   }
 
-  private lazy val nodeAddresses = new NodeAddresses(connector)
   override def getPreferredLocations(split: Partition): Seq[String] =
-    split.asInstanceOf[CassandraPartition[_, _]].endpoints.flatMap(nodeAddresses.hostNames).toSeq
+    split.asInstanceOf[CassandraPartition[_, _]].endpoints
```

Comment:
yep! So previously this got passed around as a set of hostnames which we would then map back out to a list of aliases which we knew men the same thing `nodeAddresses.hostNames`. This would make sure we would get all known interfaces for a node, just incase spark was bound on one and C* was bound on another, or public/private confusions. 

Rather than doing this over and over at the place of consumption I turned things around so instead we just keep all of that information in the partition itself rather than computing it right before we do "preferredLocations"

Tldr;
Before:  partition.oneHost ----> on read ---> mapToAllHostsIdentifiers
Now: partition.AllHostIdentIfiers